moveit_core/constraint_samplers/test/test_constraint_samplers.cpp

```diff
@@ -118,14 +118,14 @@ class LoadPlanningModelsPr2 : public testing::Test
                                                 "l_wrist_roll_link",
                                                 .01);
 
-    func_right_arm.reset(new robot_model::SolverAllocatorFn(boost::bind(&LoadPlanningModelsPr2::getKinematicsSolverRightArm, this, _1)));
-    func_left_arm.reset(new robot_model::SolverAllocatorFn(boost::bind(&LoadPlanningModelsPr2::getKinematicsSolverLeftArm, this, _1)));
+    func_right_arm = boost::bind(&LoadPlanningModelsPr2::getKinematicsSolverRightArm, this, _1);
+    func_left_arm  = boost::bind(&LoadPlanningModelsPr2::getKinematicsSolverLeftArm, this, _1);
 
     std::map<std::string, robot_model::SolverAllocatorFn> allocators;
-    allocators["right_arm"] = *func_right_arm;
-    allocators["left_arm"] = *func_left_arm;
-    allocators["whole_body"] = *func_right_arm;
-    allocators["base"] = *func_left_arm;
+    allocators["right_arm"]  = func_right_arm;
+    allocators["left_arm"]   = func_left_arm;
+    allocators["whole_body"] = func_right_arm;
+    allocators["base"]       = func_left_arm;
 
```

Comment:
There was no need for `shared_ptrs` here. The boost `boost::function` object was getting copied anyway, which ultimately comes down to copying two pointers here.